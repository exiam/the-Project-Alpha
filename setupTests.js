const Adapter = require('enzyme-adapter-react-16')

require('enzyme').configure({ adapter: new Adapter() })

require('dotenv').config({ path: '.env' })
