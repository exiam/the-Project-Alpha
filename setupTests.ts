import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
export function setup() {
  configure({ adapter: new Adapter() })
}
