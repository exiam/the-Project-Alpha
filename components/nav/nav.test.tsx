import * as React from 'react'
import Nav, { NavProps } from './nav.component'
import { setup } from '../../setupTests'
setup()
import { render, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeProps: NavProps = {
  logoText: 'Logo Text',
  items: [{ href: '', name: 'Name inside', id: 0 }],
}

describe('Nav component', () => {
  describe('rendering', () => {
    it('should render without error', () => {
      render(<Nav {...fakeProps} />)
    })

    const wrapper = render(<Nav {...fakeProps} />)

    it('should render the logo text', () => {
      expect(wrapper.find('h3').text()).toBe('Logo Text')
    })
    it('should render the links text', () => {
      expect(wrapper.find('ul li').text()).toBe('Name inside')
    })
  })
  describe('snapshot', () => {
    it('should match with the connected snapshot', () => {
      expect(
        toJSON(shallow(<Nav connected {...fakeProps} />))
      ).toMatchSnapshot()
    })
    it('should match with the not connected snapshot', () => {
      expect(toJSON(shallow(<Nav {...fakeProps} />))).toMatchSnapshot()
    })
  })
})
