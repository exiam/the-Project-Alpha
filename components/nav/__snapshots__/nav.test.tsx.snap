// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component snapshot should match with the connected snapshot 1`] = `
<navcomponent__NavContainer
  connected={true}
>
  <navcomponent__NavContainer2>
    <navcomponent__NavLogo>
      Logo Text
    </navcomponent__NavLogo>
    <navcomponent__NavItems>
      <NavItem
        connected={true}
        href=""
        id={0}
        key="0"
        name="Name inside"
      />
    </navcomponent__NavItems>
  </navcomponent__NavContainer2>
</navcomponent__NavContainer>
`;

exports[`Nav component snapshot should match with the not connected snapshot 1`] = `
<navcomponent__NavContainer
  connected={false}
>
  <navcomponent__NavContainer2>
    <navcomponent__NavLogo>
      Logo Text
    </navcomponent__NavLogo>
    <navcomponent__NavItems>
      <NavItem
        connected={false}
        href=""
        id={0}
        key="0"
        name="Name inside"
      />
    </navcomponent__NavItems>
  </navcomponent__NavContainer2>
</navcomponent__NavContainer>
`;
